pkg/cmd/server/command.go

```diff
@@ -126,4 +129,16 @@ func (cfg *Config) Complete(args []string) {
 		// if we've explicitly called out an etcd server, don't start etcd in-process
 		cfg.StartEtcd = !cfg.EtcdAddr.Provided
 	}
+
+	// if this is an all-in-one start, be sure to add our hostname to the NodeList if it is not already present
+	isAllInOne := (len(args) == 0)
```

Comment:
logically, this can use `cfg.StartMaster && cfg.StartNode` as well... if we're both a master and a node, make sure we know about ourselves